azkaban-common/src/main/java/azkaban/database/AbstractJdbcLoader.java

```diff
@@ -34,7 +36,7 @@ public AbstractJdbcLoader(final Props props) {
 
   protected Connection getDBConnection(final boolean autoCommit) throws IOException {
     Connection connection = null;
-    CommonMetrics.INSTANCE.markDBConnection();
+    SERVICE_PROVIDER.getInstance(CommonMetrics.class).markDBConnection();
```

Comment:
Consider saving this reference in the constructor? This class can be considered a dependency of this class.